src/vm/database/clarity_db.rs

```diff
@@ -201,31 +218,36 @@ impl <'a> ClarityDatabase <'a> {
     }
 
     pub fn sim_mine_blocks(&mut self, count: u32) {
-        for i in 0..count {
+        for _i in 0..count {
             self.sim_mine_block();
         }
     }
 }
 
+// this is used so that things like load_map, load_var, load_nft, etc.
+//   will throw NoSuchFoo errors instead of NoSuchContract errors.
+fn map_no_contract_as_none <T> (res: Result<Option<T>>) -> Result<Option<T>> {
+    res.or_else(|e| match e {
+        Error::Unchecked(CheckErrors::NoSuchContract(_)) => Ok(None),
+        x => Err(x)
+    })
+}
+
 // Variable Functions...
 impl <'a> ClarityDatabase <'a> {
     pub fn create_variable(&mut self, contract_identifier: &QualifiedContractIdentifier, variable_name: &str, value_type: TypeSignature) {
         let variable_data = DataVariableMetadata { value_type };
+        let key = ClarityDatabase::make_metadata_key(StoreType::VariableMeta, variable_name);
 
-        let key = ClarityDatabase::make_key_for_trip(contract_identifier, StoreType::VariableMeta, variable_name);
-
-        assert!(!self.store.has_entry(&key), format!("Clarity VM attempted to initialize existing variable '{}'", variable_name));
-
-        self.put(&key, &variable_data);
+        self.insert_metadata(contract_identifier, &key, &variable_data)
```

Comment:
Do we want to guard these instances of `insert_metadata()` with `has_metadata_entry()` like we do with contracts?

Reply:
Probably not -- the caller needs to handle the check, since some callers, like the analysis phase return a CheckError if a metadata entry already exists.